judge/views/problem.py

```diff
@@ -200,9 +200,70 @@ def get_context_data(self, **kwargs):
                                           context['description'], 'problem')
         context['meta_description'] = self.object.summary or metadata[0]
         context['og_image'] = self.object.og_image or metadata[1]
+
+        context['can_vote'] = self.object.can_vote(user)
+        # The vote this user has already cast on this problem.
+        if context['can_vote']:
+            try:
+                context['vote'] = ProblemPointsVote.objects.get(voter=user.profile, problem=self.object)
+            except ObjectDoesNotExist:
+                context['vote'] = None
+        else:
+            context['vote'] = None
```

Comment:
Is this branch required? If the user can't vote, I would hope you never access their vote.

Actually, are votes invalidated when a user is banned from voting? I'm confused either way.

Reply:
iirc, Ninjaclasher in an earlier review said something about that vote should always be defined, and that was the reason for the branch.

Reply:
Hm~ ok, leave it for now.